### ISSUE TITLE
toolset("clang") doesn't do anything

### ISSUE THREAD

**Issue by imgeself** [premake/premake-core#1507]
**What seems to be the problem?**
I am trying to use llvm toolset on windows. But when I set toolset to `clang`, premake generates projects with vs2019 toolset.

**What did you expect to happen?**
It generates visual studio project file that `PlatformToolset` set to `v142`. It should set to `ClangCL`

**What have you tried so far?**
I downloaded premake source code and modified [premake config file](https://github.com/premake/premake-core/blob/master/premake5.lua) and set the Premake5 project toolset to `clang`. It works there. But when I try on some other projects that using premake such as [Hazel](https://github.com/TheCherno/Hazel). It doesn't work there either.

**How can we reproduce this?**
I can reproduce this easily. Just create really basic workspace and project. Set toolset to clang.
```
workspace "Solution"
    configurations { "Debug", "Release" }
    startproject "Project"

project "Project"
    kind "ConsoleApp"
    toolset "clang"

```

**What version of Premake are you using?**
I tried alpha 15 and alpha 14.

**Comment by samsinsane** [comment]
> But when I set toolset to "clang", premake generates projects with vs2019 toolset.

Please elaborate.

> I downloaded premake source code and modified some of the premake config files. When I set toolset to "clang" on those files it works.

Please elaborate.

> I tried on some other projects that using premake such as Hazel. It doesn't work there either.

They provide a pre-built version of Premake, you need to go talk to them about updating their version of Premake if you're having issues with their project.

> I can reproduce this easily.

Then provide a small sample as requested.

> I tried multiple versions

Then provide the version numbers as requested.

The bug template is there to be filled in, not ignored and mostly deleted. You only delay your own help by not doing things as requested.

**Comment by imgeself** [comment]
I don't know how to elaborate more because the problem is very clear. I updated the bug report and hoping that it is more clear and understandable now.

> They provide a pre-built version of Premake, you need to go talk to them about updating their version of Premake if you're having issues with their project.

Hazel just a example, I am having this issue on my projects as well. I was trying to say that it happens other projects too.

**Comment by nickclark2016** [comment]
@imgeself I merged in code a while back to do just what you are requesting (clang option for VS2019).  I've got a custom build of premake that I use with this added.  It's PR #1462.  @samsinsane would it be worthwhile to put out a new alpha release?

**Comment by imgeself** [comment]
Thanks @nickclark2016 . I didn't know that latest alpha build doesn't have clang support for vs2019. And nobody didn't even bother to tell me that here. Indeed, it works and sets toolset to `ClangCL`. But there is another problem. When I set toolset to clang, Premake changes the formatting of links. This issue describes it #1520 .

For those who want use clang toolset with vs2019: Don't change toolset to `clang`. Use the default which is `v142`. 
And put this code on top of your premake file:
```
if _ACTION:startswith("vs") then
    require("vstudio")
    local function clangToolset(prj)
        premake.w('<PlatformToolset>ClangCL</PlatformToolset>')
    end

    premake.override(premake.vstudio.vc2010.elements, "configurationProperties", function(base, prj)
            local calls = base(prj)
            table.insert(calls, clangToolset)
            return calls
    end)
end
```
This code overrides configuration properties of all projects and set their platform toolsets to `ClangCL`. The formatting of links is going to be correct as well. Because clang uses the same compiler switches as MSVC on windows. This works like a charm and you can use this method until we get proper clang toolset support for vs2019.

**Comment by nickclark2016** [comment]
I can't say I've had an issue with clang and links, but I'll take a look.  If you want to use clang, try using msc-ClangCL if you want clang in vs2019.

**Comment by imgeself** [comment]
Thank you so much @nickclark2016 . `msc-ClangCL` works like a charm.

**Comment by nickclark2016** [comment]
No problem. Once I get off work, I'll work on a change for clang to work correctly with vs2019 generated projects

**Comment by nickclark2016** [comment]
@imgeself if you're looking for a fix, I've got one out #1522

**Comment by samsinsane** [comment]
Closing this as the original issue is resolved.

For future readers: alpha15 doesn't have Clang support for VS projects, use a master build, or a more recent release if available.
